tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	AfterAll(func() {
+		// cleanup and restore NGF instance
+		teardown(releaseName)
+		setup(getDefaultSetupCfg())
+	})
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, err
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyAndReturnNginxGateway := func(nsname types.NamespacedName) ngfAPI.NginxGateway {
+		nginxGateway, err := getNginxGateway(nsname)
+		Expect(nginxGateway).ToNot(BeNil())
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(nginxGateway.Status.Conditions).To(HaveLen(1))
+		condition := nginxGateway.Status.Conditions[0]
+
+		Expect(condition.Type).To(Equal("Valid"))
+		Expect(condition.Reason).To(Equal("Valid"))
+
+		return nginxGateway
+	}
+
+	When("testing NGF on startup", func() {
+		BeforeEach(func() {
+			teardown(releaseName)
+
+			ns := &core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		})
+
+		When("no NginxGateway resource exists", func() {
```

Comment:
Since the test framework starts in `debug` mode, that is the "default" for this test. Should I try to avoid that? Since if we choose to run the helm tests not in debug, that will break this whole test suite and just flip the logic.

Reply:
If we update the tests to not run in debug, then this test will fail and we can fix it up. So I think this is okay to leave as-is.